src/engine/Engine.ts

```diff
@@ -900,13 +901,16 @@ O|===|* >________________>\n\
     this.pageScrollPreventionMode = options.scrollPreventionMode;
 
     // initialize inputs
+    const pointerTarget = options && options.pointerScope === Input.PointerScope.Document ? document : this.canvas;
     this.input = {
       keyboard: new Input.Keyboard(),
-      pointers: new Input.Pointers(this),
+      pointers: new PointerEventReceiver(pointerTarget, this),
       gamepads: new Input.Gamepads()
     };
     this.input.keyboard.init();
-    this.input.pointers.init(options && options.pointerScope === Input.PointerScope.Document ? document : this.canvas);
+    // TODO detach?
+    // TODO rename to init?
+    this.input.pointers.attach();
```

Comment:
Will keyboard and gamepad be changed to event receiver and then `attach`? `init` seems more apt if that's its purpose/intent. `attach` implies it should be passed something to attach to (if so, maybe move `this` from ctor to `attach(this)`)

Reply:
Yah init is probably still a better name, I'll switch it back.